plugins/worker/lib/jobs.js

```diff
@@ -197,6 +212,38 @@ function stop(buildConfig) {
     );
 }
 
+/**
+ * Send message to clear cache from disk
+ * @param {Object} cacheConfig
+ */
+function clear(cacheConfig) {
+    const { id, buildClusters } = cacheConfig;
+    let queueName;
+
+    return redis
+        .hget(`${queuePrefix}buildConfigs`, id)
+        .then(data => {
+            if (data) {
+                const buildConfig = JSON.parse(data);
+
+                queueName = buildConfig.buildClusterName;
+            }
+        })
+        .then(() => {
+            if (queueName) {
+                return pushToRabbitMq({ job: 'clear', cacheConfig }, queueName, id);
+            }
+
+            if (buildClusters) {
+                return Promise.all(
+                    buildClusters.map(cluster => pushToRabbitMq({ job: 'clear', cacheConfig }, cluster, id))
+                );
+            }
```

Comment:
double publishing? is the latter just to catch all?

Reply:
if queueName os not present, we publish to all existing rabbitmq queue's

Reply:
Sorry didn’t see the return there

Reply:
by default, it should push to all queues

Reply:
@pritamstyz4ever maybe all is better in case people switch cluster back and forth...